### PR TITLE
[ACTION] Create a DocuSign Draft Envelope #5874

### PR DESCRIPTION
**Pull request by lcaresia**
### <samp>🤖 Generated by Copilot at b23d6b9</samp>

This pull request adds a new action to create draft envelopes in Docusign, and refactors the common logic for Docusign actions to support different envelope statuses. It also updates the dependencies and versions of the Docusign package and actions.

### <samp>🤖 Generated by Copilot at b23d6b9</samp>

> _To create drafts with Docusign_
> _We made a new module design_
> _It inherits from `create-signature-request`_
> _But sets a different status in the envelope_
> _And updates the action's key, name, and version_

### <samp>🤖 Generated by Copilot at b23d6b9</samp>

*  Add a new action to create draft envelopes in Docusign ([link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-d3e62b5bac0f748fb1891c4599bb0c728440ee82b76b8e91a8b3056f5de651ddR1-R16), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-01f45bc76ed852196007ab6d54d5b97396de7eb85c33f080ae3d272566f5df6bL150-R153), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-01f45bc76ed852196007ab6d54d5b97396de7eb85c33f080ae3d272566f5df6bR10-R12))
* Update the existing action to create signature requests in Docusign ([link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-af3816f92f2fe0422da4ac00b084484b19998eb310a2f70fabbba8b92ab0573cL2-R16), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-01f45bc76ed852196007ab6d54d5b97396de7eb85c33f080ae3d272566f5df6bL150-R153), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-01f45bc76ed852196007ab6d54d5b97396de7eb85c33f080ae3d272566f5df6bR10-R12))
* Update the Docusign package version in `package.json` ([link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-c9c580cce1dd9f3182dc272f6cb381007841c40b3744b780efa90e622180b066L3-R3))
* Add new entries for other components in `pnpm-lock.yaml` ([link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR634-R636), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR790-R792), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1241-R1243), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1935-R1937), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3586-R3588), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3595-R3597), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3695-R3697), [link](https://github.com/PipedreamHQ/pipedream/pull/6538/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4929-R4934))

### CODE REVIEW

File: components/docusign/actions/create-draft/create-draft.mjs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import createSignatureRequest from "../create-signature-request/create-signature-request.mjs";
+
+export default {
+  ...createSignatureRequest,
+  key: "docusign-create-draft",
+  version: "0.0.1",
+  name: "Create Draft",
+  description: "Creates and sends an envelope or creates a draft envelope. [See the docs here](https://developers.docusign.com/docs/esign-rest-api/reference/envelopes/envelopes/create/)",
```
Comment (dylburger) [inline components/docusign/actions/create-draft/create-draft.mjs:8]:
```suggestion
  description: "Create and send an envelope or create a draft envelope. [See the documentation here](https://developers.docusign.com/docs/esign-rest-api/reference/envelopes/envelopes/create/)"
```

Comment (dylburger) [reply components/docusign/actions/create-draft/create-draft.mjs:8]: Changed 'Creates' to 'Create', 'sends' to 'send', and 'docs' to 'documentation' to follow Google's Developer docs style guide (use present tense and prefer 'documentation' over 'docs')

**Review by luancazarine — APPROVED** [review]
Hi @lcaresia, thanks for your contribution! LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[DocuSign_Draft_Envelope_5874_1035.pdf](https://github.com/PipedreamHQ/pipedream/files/11513241/DocuSign_Draft_Envelope_5874_1035.pdf)

File: components/docusign/actions/create-draft/create-draft.mjs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import createSignatureRequest from "../create-signature-request/create-signature-request.mjs";
+
+export default {
+  ...createSignatureRequest,
+  key: "docusign-create-draft",
+  version: "0.0.1",
+  name: "Create Draft",
+  description: "Create and send an envelope or create a draft envelope. [See the documentation here](https://developers.docusign.com/docs/esign-rest-api/reference/envelopes/envelopes/create/)",
```
Comment (dylburger) [inline components/docusign/actions/create-draft/create-draft.mjs:8]:
```suggestion
  description: "Create and send an envelope, or create a draft envelope. [See the documentation here](https://developers.docusign.com/docs/esign-rest-api/reference/envelopes/envelopes/create/)",
```

Comment (dylburger) [reply components/docusign/actions/create-draft/create-draft.mjs:8]: Added a comma for clarity, Google's Developer docs style guide: Clarity

**Review by luancazarine — APPROVED** [review]

**Pull request merged**